source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -23,7 +23,10 @@ StreamHandleWrapper::StreamHandleWrapper(Filters::Common::Lua::Coroutine& corout
         if (state_ == State::Running) {
           throw Filters::Common::Lua::LuaException("script performed an unexpected yield");
         }
-      }) {}
+      }),
+      fireAndForgetHttpWriter_(new FireAndForgetHttpWriter(filter)) {}
+
+StreamHandleWrapper::~StreamHandleWrapper() { delete fireAndForgetHttpWriter_; }
```

Comment:
if you are going to be doing heap allocations please use std::unique_ptr/std::shared_ptr to manage the memory